tests/composite_pk/test_generic.py

```diff
@@ -0,0 +1,302 @@
+import json
+from datetime import date, datetime
+from uuid import UUID
+
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.contenttypes.prefetch import GenericPrefetch
+from django.core.serializers.json import DjangoJSONEncoder
+from django.db import connection
+from django.db.models import Count
+from django.test import TestCase, skipIfDBFeature, skipUnlessDBFeature
+
+from .models import Comment, Dummy, Post, Tag, Tenant, User
+
+
+class CompositePKGenericTests(TestCase):
+    POST_1_ID = "e1516ac0-4469-4306-b0ac-2c435e677aa4"
+    DUMMY_1_UUID = "81598d5a-fa03-4800-bae4-823ca12824d5"
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.dummy_1 = Dummy.objects.create(
+            small_integer=32767,
+            integer=2147483647,
+            big_integer=9223372036854775807,
```

Comment:
Do we handle negative numbers correctly too?

Reply:
I added a test for it 👍